src/main/java/io/vertx/ext/mongo/impl/config/CompressorListParser.java

```diff
@@ -0,0 +1,77 @@
+package io.vertx.ext.mongo.impl.config;
+
+import com.mongodb.ConnectionString;
+import com.mongodb.MongoCompressor;
+import com.mongodb.lang.Nullable;
+import io.vertx.core.json.JsonObject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CompressorListParser {
+
+  private final List<String> compressorKeys;
+  private final ConnectionString connectionString;
+  private final JsonObject config;
+
+  public CompressorListParser(ConnectionString connectionString, JsonObject config) {
+    this.connectionString = connectionString;
+    this.config = config;
+
+    this.compressorKeys = new ArrayList<>();
+    this.compressorKeys.add("compressors");
+    this.compressorKeys.add("zlibcompressionlevel");
+  }
+
+  public List<MongoCompressor> compressorList() {
+    String compressors = "";
+    Integer zlibCompressionLevel = null;
+
+    for (final String key : compressorKeys) {
+      String value = config.getString(key);
+      if (value == null) {
+        continue;
+      }
+
+      if (key.equals("compressors")) {
+        compressors = value;
+      } else if (key.equals("zlibcompressionlevel")) {
+        zlibCompressionLevel = Integer.parseInt(value);
+      }
+    }
+
+    List<MongoCompressor> connStrCompressors = connectionString.getCompressorList();
+    if (compressors.trim().length() == 0) {
+      return connStrCompressors;
+    } else {
+      List<MongoCompressor> compressorList = buildCompressors(compressors, zlibCompressionLevel);
+      if (connStrCompressors != null) {
+        compressorList.addAll(connStrCompressors);
+      }
+      return compressorList;
+    }
+  }
+
+  private List<MongoCompressor> buildCompressors(final String compressors, @Nullable final Integer zlibCompressionLevel) {
+    List<MongoCompressor> compressorsList = new ArrayList<>();
+
+    for (String cur : compressors.split(",")) {
```

Comment:
We could be more tolerant and trim+lowercase each compressor name provided by users